### DISCUSSION TITLE
color scheme

### DISCUSSION THREAD

**Discussion by Taxom** [jvde-github/AIS-catcher#119] [Q&A] [answered]
Is it possible to configure the color scheme of the web interface using plugins? I want to make a dark theme more calm (github dark default style), not so contrasting.
Thanks!
Max.

**Comment by jvde-github** [comment]
Yes. Just make a file like this:
```
        .dark {
            --primary-color: pink;
            --secondary-color: pink;
            --tertiary-color: #0b1033;
            --panel-color: #f2f8ff;
            --ok-color: rgb(129, 244, 129);
            --error-color: red;
            --header-color: var(--primary-color);
            --header-title-color: #0e6cd8;
            --selection-color: orange;
            --mainspace-color: white;
            --backdrop-color: #f5f7fa;
            --focus-color: red;
            --light-gray: lightgray;
            --header-font-color: white;
            --menu-background-color: white;
            --menu-background-hover-color: #daebfe;
            --menu-font-color: rgb(66, 66, 66);
            --mainspace-container-color: lightgrey;
            --table-header-background-color: var(--primary-color);
            --table-background-color: white;
            --table-selection-background-color: #daebfe;
            --table-alternate-background-color: var(--panel-color);
            --main-font-color: var(--tertiary-color);
            --shipcard-footer-background-color: var(--tertiary-color);
            --shipcard-footer-hover-color: var(--primary-color);
            --shipcard-selection-color: rgb(240, 241, 248);
            --shipcard-bottom-border-color: rgb(217, 218, 218);
            --chart1-color: rgba(93, 169, 213, 0.9);
            --chart2-color: rgba(157, 93, 213,0.9);
            --chart3-color: rgba(93, 213, 97,0.9);
            --chart4-color: rgba(213, 157, 93,0.9);
            --chart5-color: rgba(93, 213, 154,0.9);
            --hover-transform: scale(0.93);
            --hover-transition: transform ease-out 0.2s, background 0.4s;
            --hover-background: rgba(255, 255, 255, 0.5);
            --graph-header-color: white;
            --tooltip-background-color: #daebfe;
        }
```
If it is called colors.txt, you can include it as this:
```
AIS-catcher.exe ... -N 8201 STYLE colors.txt
```

**Comment by Taxom** [comment]
Thank you very much!
I'm experimenting with this and I've managed to achieve some results. I haven't configured everything yet, but it's already better.
Best regards,
Maxim.

`
.dark {
  --primary-color: #3c444d;
  --secondary-color: #6a737d;
  --tertiary-color: #202326;
  --panel-color: #1b1f23;
  --ok-color: #28a745;
  --error-color: #d73a49;
  --header-color: var(--primary-color);
  --header-title-color: #6f6b6bad;
  --selection-color: #ffd33d;
  --mainspace-color: #595d63;
  --backdrop-color: #f5f7fa;
  --focus-color: #f9826c;
  --light-gray: #e1e4e8;
  --header-font-color: #ffffff;
  --menu-background-color: #0d1117;
  --menu-background-hover-color: #30363d;
  --menu-font-color: #dae2eb;
  --mainspace-container-color: #fafbfc;
  --table-header-background-color: #24292e;
  --table-background-color: var(--mainspace-color);
  --table-selection-background-color: #ffd33d;
  --table-alternate-background-color: var(--panel-color);
  --main-font-color: #b6b7bb;
  --shipcard-footer-background-color: #f0f6fc;
  --shipcard-footer-hover-color: #24292e;
  --shipcard-selection-color: #f0f6fc;
  --shipcard-bottom-border-color: #c9d1d9;
  --chart1-color: rgba(93, 169, 213, 0.9);
  --chart2-color: rgba(157, 93, 213, 0.9);
  --chart3-color: rgba(93, 213, 97, 0.9);
  --chart4-color: rgba(213, 157, 93, 0.9);
  --chart5-color: rgba(93, 213, 154, 0.9);
  --hover-transform: scale(0.93);
  --hover-transition: transform ease-out 0.2s, background 0.4s;
  --hover-background: rgba(255, 255, 255, 0.5);
  --graph-header-color: #0d0e0f;
  --tooltip-background-color: #30363d;
}

`

**Comment by Taxom** [comment]
Hello, I can't find the variable that controls the color of the grid and graph rings.


![Screenshot 2023-06-08 at 12-30-08 AIS-catcher](https://github.com/jvde-github/AIS-catcher/assets/20313019/cd4a35a1-7d6e-456f-84ee-c931ca834e3e)
![Screenshot 2023-06-08 at 12-27-48 AIS-catcher](https://github.com/jvde-github/AIS-catcher/assets/20313019/14253514-1319-4ac2-af9e-b4e08e135b50)


this is my colors.txt:
//
.dark {
  --primary-color: #3c444d;
  --secondary-color: #6a737d;
  --tertiary-color: #202326;
  --panel-color: #1b1f23;
  --ok-color: #28a745;
  --error-color: #d73a49;
  --header-color: var(--primary-color);
  --header-title-color: #6f6b6bad;
  --selection-color: #ffd33d;
  --mainspace-color: #595d63;
  --backdrop-color: #000;
  --focus-color: #f9826c;
  --light-gray: #e1e4e8;
  --header-font-color: #ffffff;
  --menu-background-color: #0d1117;
  --menu-background-hover-color: #30363d;
  --menu-font-color: #dae2eb;
  --mainspace-container-color: #000;
  --table-header-background-color: #24292e;
  --table-background-color: var(--mainspace-color);
  --table-selection-background-color: #5e325d;
  --table-alternate-background-color: var(--panel-color);
  --main-font-color: #b6b7bb;
  --shipcard-footer-background-color: #39536c;
  --shipcard-footer-hover-color: #24292e;
  --shipcard-selection-color: ##444546;
  --shipcard-bottom-border-color: #c9d1d9;
  --chart1-color: rgba(93, 169, 213, 0.9);
  --chart2-color: rgba(157, 93, 213, 0.9);
  --chart3-color: rgba(93, 213, 97, 0.9);
  --chart4-color: rgba(213, 157, 93, 0.9);
  --chart5-color: rgba(93, 213, 154, 0.9);
  --hover-transform: scale(0.93);
  --hover-transition: transform ease-out 0.2s, background 0.4s;
  --hover-background: rgba(255, 255, 255, 0.5);
  --graph-header-color: #0d0e0f;
  --tooltip-background-color: #30363d;
}
//


Thank you,
Maxim.

**Reply by jvde-github** [reply]
Unfortunately, these cannot be controlled yet. Also, I remember that I have to fix the colors in the graphs when switching between dark and light modus. This does not yet automatically flow.

**Comment by Taxom** [comment, marked as answer]
Thanks for [fix dark mode for charts](https://github.com/jvde-github/AIS-catcher/commit/c34b98a1d32d89539fc5faaf7c7cc2e2bdd78b6c) got much better.
![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/cd546c5a-bd9a-4888-a0ee-e0d607807250)
